### PR TITLE
Added code format check of `.proto` files

### PR DESCRIPTION
**Pull request by ArthurBandaryk**
Added `check_style_proto` in the `tools` directory. It's a simple bash script
which is run in the check_format_code.yaml action.

### CODE REVIEW

**Comment by ArthurBandaryk** [comment]
Also tested `check_style_proto` script locally. Was playing with correct format, then added some format errors and made some lines exceed `80` characters. The results and `test.proto` file are on the screen below.
Screen Shot 2021-09-30 at 9 40 15 PM
Screen Shot 2021-09-30 at 9 41 04 PM
Screen Shot 2021-09-30 at 9 41 37 PM

**Review by while-false — CHANGES_REQUESTED** [review]
Hey Arthur, thanks for executing on this! I like the setup with the shell script at the `prototools` linter. I'd like to `clang-format` to the mix.

`clang-format` can also be used to check and enforce protobuf files, and since we are already using it for C++ files, it is easy to expand the `.clang-format` file: all you'd need to add is a segment like
```
---
Language: Proto
BasedOnStyle: Google
ColumnLimit: 0
```

Then `.proto` files can be checked with:

```
clang-format --style=file --Werror ${file}
```

and your checks should be doable with something like:

```bash
if ! clang-format --style=file --Werror ${file}; then
   # handle error case
fi
```


Give it a shot! Ask if anything is unclear and looking forward to seeing the next review. :)

File: tools/check_style_proto [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    # prototool format [file] -l (or --lint)
+    # but in fact it only prints that Format returned a diff.
+    # Actually if the file is well formatted the command above will
+    # return 0. So I used this fact in the code below!
+    # The link to the official doc:
+    # https://github.com/uber/prototool#prototool-format
+    # Read the Command Overview section!
+
+    printf "Checking ${file:2} ...\n"
+    prototool format -l ${file:2}
```
[the remark this answers is not in the document]
Comment (ArthurBandaryk) [reply tools/check_style_proto:43]: Yep, I will omit the first `./`.

File: tools/check_style_proto [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    format_status=$(echo $?)
+
+    if [[ ${format_status} != 0 ]]; then
+      tput bold # Bold text in terminal
+      tput setaf 1 # Red font in terminal
+      printf "${file:2} file has incorrect code format!\n"
```
[the remark this answers is not in the document]
Comment (ArthurBandaryk) [reply tools/check_style_proto:49]: Yes, I will do omit.

File: tools/check_style_proto [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  if [[ ${#proto_file_paths[@]} == 0 ]]; then
+    printf "There are no .proto files to check!\n"
+    exit 0
+  fi  
+
+  status_exit=0
+
+  for file in ${proto_file_paths}
+  do
+    # Check if the specific file has correct code format
```
[the remark this answers is not in the document]
Comment (ArthurBandaryk) [reply tools/check_style_proto:28]: I will move it up.

**Comment by ArthurBandaryk** [comment]
1. Need advise from Ben to choose how we would like to check `.proto` files: using `prototool` or `clang-format`.
2. For now ,as you could remember, `jidicula's clang-format-action` doesn't check if `line[I] > 80` chars. So actually I'm thinking about to remove this action and to base on the Ben's script `tools/pre-commit`. Any thoughts guys?

**Pull request closed without merging**